src/Listener/OpenTracker.php

```diff
@@ -47,8 +47,13 @@ public function onCompose(ComposeBatchEvent $e) {
       $mailParams = $task->getMailParams();
 
       if (!empty($mailParams) && !empty($mailParams['html'])) {
+        $tracking_url = $config->userSystem->languageNegotiationURL(
+          $config->userFrameworkBaseURL . \Civi::settings()->get('userFrameworkResourceURL') . "/extern/open.php?q=" . $task->getEventQueueId(),
+          FALSE,
+          TRUE
+        );
```

Comment:
@jensschuppe Oh, yeah, this can be tricky! I'm glad you were able to get a patch that worked on your system.

FWIW, path and URL settings (like `userFrameworkResourceURL`) can be stored in different formats, depending on the history/preferences of the particular site. The setting might have:

* A relative URL. In this case, the code above probably works.
* An absolute URL. In this case, the code probably fails.
* A variable URL expression (like `[civicrm.root]/foo` or `[cms.root]/bar`). In this case, the code fails. On my system, it produces a URL like:

    ```
    http://dmaster.bknix:8001/[civicrm.root]//extern/open.php?q=123
    ```

As a rule of thumb, `Civi::settings()` will give you the raw/unfiltered value of `userFrameworkResourceURL`, and `Civi::paths()` can convert that to a usable path or URL. Ex:

```php
$rawSetting = Civi::settings()->get('userFrameworkResourceURL');
$fullUrl = Civi::paths()->getUrl($rawSetting);

// Fun fact: $config->userFrameworkResourceURL is basically doing those two steps.
```

FWIW, in the past few releases of `civicrm-core`, there have been some changes affecting the open-tracker URL. Most notable is the addition of [CRM_Utils_System::externUrl(...)](https://github.com/civicrm/civicrm-core/blob/5.26/CRM/Utils/System.php#L301-L306) in v5.23+. It's sort analogous to `CRM_Utils_System::url()`, but... it's for `extern`. 😃 It's using a slightly different formula by default, and it exposes a hook so that others can change the formula.

Switching to `System::externUrl(...)` should, at a minimum, get the ball out of `flexmailer`'s court (putting `flexmailer` in the same boat as the MailingBAO engine). See also: #56 

Hopefully that also fixes dev/mail#17. (But I can't test -- don't have the affected configuration available.) If you have a chance to retest at some point, that'd be great.